web/src/user_profile.js

```diff
@@ -321,6 +353,39 @@ export function get_custom_profile_field_data(user, field, field_types) {
     return profile_field;
 }
 
+function render_bot_profile_fields(bot) {
+    const user_type = {
+        name: bot.full_name,
+        should_add_guest_user_indicator: bot.is_guest,
+    };
+    $(".user_profile_name").html(render_user_full_name(user_type));
+
+    const $bot_owner_field = $(".bot_owner_user_field");
+    $bot_owner_field.attr("data-field-id", bot.bot_owner_id);
+
+    $("#avatar").css("background-image", `url(${people.medium_avatar_url_for_person(bot)})`);
+}
+
+export function update_user_custom_profile_fields(user) {
+    if (!(modals.any_active() && modals.active_modal() === "#user-profile-modal")) {
+        return;
+    }
+    const $custom_profile_field = $("#content");
+    const field_types = realm.custom_profile_field_types;
+
+    const profile_fields = realm.custom_profile_fields
+        .map((f) => get_custom_profile_field_data(user, f, field_types))
+        .filter((f) => f.name !== undefined);
+
+    const profile_data = {profile_fields};
+    $custom_profile_field.html(render_user_custom_profile_fields(profile_data));
+    custom_profile_fields_ui.initialize_custom_user_type_fields(
+        "#user-profile-modal #content",
+        user.user_id,
+        false,
+    );
+}
```

Comment:
This ensures that the mentor pills are properly rendered. Is this fine?

Reply:
I guess this is the easiest way to avoid complexity, so should be fine from my side. Will see if Tim has any other suggestions when he reviews.